### PR TITLE
Base structure of estimate Service and dependency service jar and model class

### PR DESCRIPTION
**Pull request by pintu-eGov**

### CODE REVIEW

File: backend/estimate-service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.egov</groupId>
```
Comment (rushang7-eGov) [inline backend/estimate-service/pom.xml:4]: Make this `org.egov.works`. 
Make the same change in other files as well.

File: backend/estimate-service/pom.xml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (rushang7-eGov) [inline backend/estimate-service/pom.xml:1]: Has this passed the snyk vulnerability check?

File: backend/estimate-service/pom.xml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    <properties>
+        <java.version>1.8</java.version>
+        <maven.compiler.source>${java.version}</maven.compiler.source>
+        <maven.compiler.target>${java.version}</maven.compiler.target>
+        <postgresql.version>42.3.3</postgresql.version>
+    </properties>
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>1.5.9.RELEASE</version>
```
Comment (rushang7-eGov) [inline backend/estimate-service/pom.xml:18]: Upgrade to Spring 2. 
Check the tracer library for `minor` version compatibility.

File: backend/estimate-service/pom.xml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        <dependency>
+            <groupId>io.swagger</groupId>
+            <artifactId>swagger-core</artifactId>
+            <version>1.5.18</version>
+        </dependency>
+        <!-- Egov dependencies -->
+        <dependency>
+            <groupId>org.egov.services</groupId>
+            <artifactId>tracer</artifactId>
+            <version>1.1.3-SNAPSHOT</version>
```
Comment (rushang7-eGov) [inline backend/estimate-service/pom.xml:65]: Use `2.0.0-SNAPSHOT`

File: backend/estimate-service/pom.xml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            <artifactId>mdms-client</artifactId>
+            <version>0.0.2-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <scope>runtime</scope>
+            <version>${postgresql.version}</version>
+        </dependency>
+        <dependency>
```
Comment (rushang7-eGov) [inline backend/estimate-service/pom.xml:93]: Remove unused dependencies. 
Check this for all the dependencies.

File: backend/estimate-service/pom.xml [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            <groupId>org.jsoup</groupId>
+            <artifactId>jsoup</artifactId>
+            <version>1.10.2</version>
+        </dependency>
+        <dependency>
+            <groupId>org.flywaydb</groupId>
+            <artifactId>flyway-core</artifactId>
+        </dependency>
+    </dependencies>
+    <repositories>
```
Comment (rushang7-eGov) [inline backend/estimate-service/pom.xml:103]: Verify if the provided urls work. 
`egovernments.org` is changed to `digit.org`. 
Copy from [here](https://github.com/misdwss/iFix-Dev/blob/develop/domain-services/fiscal-event-service/pom.xml#L113)

File: backend/estimate-service/pom.xml (hunk 1 above)
Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:4]: done.

File: backend/estimate-service/pom.xml (hunk 4 above)
Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:65]: done.

File: backend/estimate-service/pom.xml (hunk 6 above)
Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:103]: done.

File: backend/estimate-service/pom.xml (hunk 5 above)
Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:93]: This is required while we are performing a search.  have rechecked the unused dependencies

File: backend/estimate-service/pom.xml (hunk 3 above)
Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:18]: I have upgraded it to "2.2.6.RELEASE" version of Spring boot (same as tracer library's version)

File: backend/estimate-service/pom.xml (hunk 2 above)
Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:1]: yes.

Comment (rushang7-eGov) [reply backend/estimate-service/pom.xml:1]: Doesn't this version of spring have the log4j vulnerability issue?

Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:1]: no. But its better to add this "<log4j2.version>2.17.1</log4j2.version>" dependency. Shall i go ahead?

File: backend/estimate-service/src/main/java/org/egov/works/web/models/SaveEstimate.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+ */
+@Validated
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2022-07-27T14:01:35.043+05:30")
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class SaveEstimate {
```
Comment (rushang7-eGov) [inline backend/estimate-service/src/main/java/org/egov/works/web/models/SaveEstimate.java:18]: Why this separate class? We don't need to create separate classes for kafka topic objects. 
Same applies for update estimate.

File: backend/estimate-service/src/main/java/org/egov/works/util/ResponseHeaderCreator.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import org.egov.works.web.models.APIInfo;
+import org.egov.works.web.models.RequestHeader;
+import org.egov.works.web.models.ResponseHeader;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class ResponseHeaderCreator {
+
+    public ResponseHeader createResponseHeaderFromRequestHeader(final RequestHeader requestHeader, final Boolean success) {
```
[the remark this answers is not in the document]
Comment (pintu-eGov) [reply backend/estimate-service/src/main/java/org/egov/works/util/ResponseHeaderCreator.java:13]: I'll change the method name.

File: backend/estimate-service/src/main/java/org/egov/works/web/models/SaveEstimate.java (hunk 7 above)
Comment (pintu-eGov) [reply backend/estimate-service/src/main/java/org/egov/works/web/models/SaveEstimate.java:18]: These were mentioned in the swagger doc [here](https://app.swaggerhub.com/apis/eGovTrial/Estimate-Service/1.0.0#/save-estimate.) hence while generating the project these objects got created.

File: backend/estimate-service/src/main/java/org/egov/works/web/models/APIInfo.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+@ApiModel(description = "Info of the API being called")
+@Validated
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2022-07-27T14:01:35.043+05:30")
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class APIInfo {
```
[the remark this answers is not in the document]
Comment (pintu-eGov) [reply backend/estimate-service/src/main/java/org/egov/works/web/models/APIInfo.java:20]: These all are generated while generating the spring boot project from swagger file.

[the remark this answers is not in the document]
Comment (rushang7-eGov) [reply backend/estimate-service/src/main/java/org/egov/works/web/models/APIInfo.java:20]: Okay. But now do as I am instructing. Delete these classes and use the classes available from service-common.

[the remark this answers is not in the document]
Comment (pintu-eGov) [reply backend/estimate-service/src/main/java/org/egov/works/web/models/APIInfo.java:20]: done

File: backend/estimate-service/pom.xml [hunk 10]
```diff
@@ -64,6 +65,16 @@
             <artifactId>tracer</artifactId>
             <version>2.0.0-SNAPSHOT</version>
         </dependency>
+        <dependency>
+            <groupId>org.egov.services</groupId>
+            <artifactId>services-common</artifactId>
+            <version>1.0.1-SNAPSHOT</version>
+        </dependency>
+        <dependency>
```
Comment (rushang7-eGov) [inline backend/estimate-service/pom.xml:73]: Don't add this. Only services-common should be enough

File: backend/estimate-service/pom.xml (hunk 2 above)
Comment (rushang7-eGov) [reply backend/estimate-service/pom.xml:1]: Yes. Add it

File: backend/estimate-service/src/main/java/org/egov/works/web/models/workflow/BusinessService.java [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@ToString
+@EqualsAndHashCode(of = {"tenantId","businessService"})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class BusinessService   {
```
Comment (rushang7-eGov) [inline backend/estimate-service/src/main/java/org/egov/works/web/models/workflow/BusinessService.java:31]: Are these files present in projects like pgr and pt? Or have they used it from some library? 
There is a new library being developed. Look if it has any reusable classes. 
[digit-models](https://github.com/egovernments/DIGIT-Dev/tree/develop/core-services/libraries/digit-models)

**Review by rushang7-eGov — APPROVED** [review]

File: backend/estimate-service/src/main/java/org/egov/works/web/models/workflow/BusinessService.java (hunk 11 above)
Comment (pintu-eGov) [reply backend/estimate-service/src/main/java/org/egov/works/web/models/workflow/BusinessService.java]: done.

File: backend/estimate-service/pom.xml (hunk 10 above)
Comment (pintu-eGov) [reply backend/estimate-service/pom.xml:73]: done

**Pull request merged**